modules/functional-test/src/main/java/com/intuit/wasabi/tests/service/RapidExperimentationTest.java

```diff
@@ -0,0 +1,328 @@
+        assignment = putAssignment(exp, assignment, new User("user6"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+
+        // Assignment assignment2 = postAssignment(exp, new User("user7"), "PROD");
+        // assertReturnCode(response, HttpStatus.SC_OK);
+        // Assert.assertEquals(assignment2.status, "EXPERIMENT_PAUSED");
+
+        assignment = putAssignment(exp, assignment, new User("user8"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+
+        assignment = putAssignment(exp, assignment, new User("user9"), "PROD");
+        assertReturnCode(response, HttpStatus.SC_OK);
+        exp = getExperiment(exp);
+        Assert.assertEquals(exp.state, Constants.EXPERIMENT_STATE_PAUSED);
+        expList.set(2, rapidExperiment3);
+    }
+
+    /**
+     * This test case covers a scenario where we do batch assignment for a group of experiments that is a combination of
+     * rapid experiments and normal experiments. It validates the behavior when we do bulk assignment of a user to that
+     * batch
+     */
+    @Test
+    public void testAssignmentForBatchWithFewRapidExperiments() {
+
+        Page partialpage = new Page(PARTIAL_BATCH_PAGE, true);
+        // lets create experiments for the batch
+        for (int i = 1; i <= NUMBER_OF_EXPERIMENTS_PER_BATCH; i++) {
+            Experiment exp = ExperimentFactory.createExperiment("_" + UUID.randomUUID().toString(), -1)
+                    .setApplication(application);
+
+            // lets make first two experiments in the batch as rapid experiments
+            if (i <= NUMBER_OF_RAPID_EXPERIMENTS_IN_BATCH) {
+                exp.setIsRapidExperiment(true).setUserCap(RAPID_EXP_MAX_USERS);
+            }
+            exp = postExperiment(exp);
+            List<Bucket> bucketList = BucketFactory.createBuckets(exp, 3);
+            postBuckets(bucketList);
+
+            exp.state = Constants.EXPERIMENT_STATE_RUNNING;
+            exp = putExperiment(exp);
+            batchExperiments.add(exp);
+
+        }
+        assignPageToExperimentsList(batchExperiments, partialpage);
+        // the assignment will happen until the max users cap is reached for all experiments in batch
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            List<Assignment> assignments = postAssignments(application, partialpage, new User("user" + i));
+            for (Assignment assignment : assignments) {
+                Assert.assertEquals(assignment.status, "NEW_ASSIGNMENT");
+
+            }
+        }
+
+        // if we do one more assignment the rapid experiments should change to PAUSED/STOPPED state and the remaining
+        // should be in running state
+        List<Assignment> assignments = postAssignments(application, partialpage, new User("user1000"));
+        for (int i = 1; i <= RAPID_EXP_MAX_USERS; i++) {
+            Experiment exp = getExperiment(batchExperiments.get(i - 1));
+            if (i <= NUMBER_OF_RAPID_EXPERIMENTS_IN_BATCH) {
```

Comment:
@sudheerkethireddy Instead of basing this check on the position of the experiment in the list, you could check whether the experiment is a rapid experiment or not and based on that, do the asserts.